homeassistant/components/light/xiaomi_miio.py

```diff
@@ -64,16 +87,19 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
             light = PhilipsEyecare(host, token)
             device = XiaomiPhilipsEyecareLamp(name, light, device_info)
             devices.append(device)
+            hass.data[PLATFORM][host] = device
         elif device_info.model == 'philips.light.ceiling':
             from miio import Ceil
             light = Ceil(host, token)
             device = XiaomiPhilipsCeilingLamp(name, light, device_info)
             devices.append(device)
+            hass.data[PLATFORM][host] = device
         elif device_info.model == 'philips.light.bulb':
             from miio import PhilipsBulb
             light = PhilipsBulb(host, token)
             device = XiaomiPhilipsLightBall(name, light, device_info)
             devices.append(device)
+            hass.data[PLATFORM][host] = device
         else:
             _LOGGER.error(
                 'Unsupported device found! Please create an issue at '
```

Comment:
return after the error and add the device on one point to `hass.data` date make the code simple

Reply:
Done.